### PR TITLE
Dockerfile: Check ROS_VERSION when adding repos

### PR DESCRIPTION
**Pull request by yaymalaga**
This fixes library incompatibilities while installing dependencies

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -18,10 +18,13 @@ ENV ROS_DISTRO=${ROS_DISTRO}
...
-    curl -sSL https://raw.githubusercontent.com/ros/rosdistro/master/ros.key -o /usr/share/keyrings/ros-archive-keyring.gpg && \
-    echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/ros-archive-keyring.gpg] http://packages.ros.org/ros2/ubuntu $(. /etc/os-release && echo $UBUNTU_CODENAME) main" | tee /etc/apt/sources.list.d/ros2.list > /dev/null && \
+    if [[ "$ROS_VERSION" == "1" ]]; then \
+        apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys C1CF6E31E6BADE8868B172B4F42ED6FBAB17C654 && \
+        echo "deb http://packages.ros.org/ros/ubuntu focal main" > /etc/apt/sources.list.d/ros1-latest.list ; \
+    elif [[ "$ROS_VERSION" == "2" ]]; then \
+    	curl -sSL https://raw.githubusercontent.com/ros/rosdistro/master/ros.key -o /usr/share/keyrings/ros-archive-keyring.gpg && \
+    	echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/ros-archive-keyring.gpg] http://packages.ros.org/ros2/ubuntu $(. /etc/os-release && echo $UBUNTU_CODENAME) main" | tee /etc/apt/sources.list.d/ros2.list > /dev/null ; \
+    fi && \
```
Comment (jpbusch) [inline docker/Dockerfile:27]:
```suggestion
    if [[ "$ROS_DISTRO" == "noetic" ]]; then \
        apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys C1CF6E31E6BADE8868B172B4F42ED6FBAB17C654 && \
        echo "deb http://packages.ros.org/ros/ubuntu focal main" > /etc/apt/sources.list.d/ros1-latest.list ; \
    else \
    	curl -sSL https://raw.githubusercontent.com/ros/rosdistro/master/ros.key -o /usr/share/keyrings/ros-archive-keyring.gpg && \
    	echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/ros-archive-keyring.gpg] http://packages.ros.org/ros2/ubuntu $(. /etc/os-release && echo $UBUNTU_CODENAME) main" | tee /etc/apt/sources.list.d/ros2.list > /dev/null ; \
    fi && \
```

File: docker/Dockerfile [hunk 2]
```diff
@@ -137,10 +140,13 @@ ENV ROS_DISTRO=${ROS_DISTRO}
...
```
Comment (jpbusch) [inline docker/Dockerfile:149]:
```suggestion
    if [[ "$ROS_DISTRO" == "noetic" ]]; then \
        apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys C1CF6E31E6BADE8868B172B4F42ED6FBAB17C654 && \
        echo "deb http://packages.ros.org/ros/ubuntu focal main" > /etc/apt/sources.list.d/ros1-latest.list ; \
    else \
    	curl -sSL https://raw.githubusercontent.com/ros/rosdistro/master/ros.key -o /usr/share/keyrings/ros-archive-keyring.gpg && \
    	echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/ros-archive-keyring.gpg] http://packages.ros.org/ros2/ubuntu $(. /etc/os-release && echo $UBUNTU_CODENAME) main" | tee /etc/apt/sources.list.d/ros2.list > /dev/null ; \
    fi && \
```

**Review by jpbusch — CHANGES_REQUESTED** [review]
Thank you for your contribution, @yaymalaga!

Since `docker-ros` is also supposed to work with base images in which the `ROS_VERSION` variable is not set and `noetic` is the only non-EOL distro for ROS 1, I would suggest checking only for `ROS_DISTRO`, which implies `ROS_VERSION`.

Apart from that, it would be interesting to know which incompatibilities have occurred in your case. Could you please describe the problem briefly?

**Comment by jpbusch** [comment]
Could you please also modify the GitHub actions `.github/workflows` such that the checks also run on PRs? Apparently we have set it up incorrectly, as no checks are currently being tested in this PR.

Simply change the `on` key to this:

```yaml
on: [push, pull_request]
```

**Comment by yaymalaga** [comment]
> Since `docker-ros` is also supposed to work with base images in which the `ROS_VERSION` variable is not set and `noetic` is the only non-EOL distro for ROS 1, I would suggest checking only for `ROS_DISTRO`, which implies `ROS_VERSION=1`.

That's sound like a good option yeah!

> Apart from that, it would be interesting to know which incompatibilities have occurred in your case. Could you please describe the problem briefly?

I was having unmet dependencies related to Gazebo, as shown below, which simply disappeared after remove the ROS1 repos:
```bash
#27 469.5  gazebo : Depends: libgazebo11 (= 11.10.2+dfsg-1) but 11.11.0-1~focal is to be installed
#27 469.5  libgazebo-dev : Depends: libignition-transport8-dev but it is not going to be installed
#27 469.5                  Depends: libignition-msgs5-dev (>= 5.1) but it is not going to be installed
#27 469.5                  Depends: libignition-common3-dev (>= 3.14.0+dfsg-3) but it is not going to be installed
#27 469.5                  Depends: libignition-fuel-tools4-dev but it is not going to be installed
#27 469.5                  Depends: libgazebo11 (= 11.10.2+dfsg-1) but 11.11.0-1~focal is to be installed
#27 469.5  libgazebo11 : Depends: libboost-filesystem1.71.0 but it is not installable
#27 469.5                Depends: libboost-iostreams1.71.0 but it is not installable
#27 469.5                Depends: libboost-program-options1.71.0 but it is not installable
#27 469.5                Depends: libboost-regex1.71.0-icu66 but it is not installable
#27 469.5                Depends: libboost-thread1.71.0 but it is not installable
#27 469.5                Depends: libbullet2.88 (>= 2.88+dfsg) but it is not installable
#27 469.5                Depends: libdart6 (>= 6.9.2) but it is not installable
#27 469.5                Depends: libdart6-collision-bullet (>= 6.9.2) but it is not installable
#27 469.5                Depends: libgdal26 (>= 3.0.0) but it is not installable
#27 469.5                Depends: libignition-fuel-tools4 (>= 4.4.0) but it is not going to be installed
#27 469.5                Depends: libignition-msgs5 (>= 5.9.0) but it is not going to be installed
#27 469.5                Depends: libignition-transport8 (>= 8.2.1) but it is not going to be installed
#27 469.5                Depends: libprotobuf17 but it is not installable
#27 469.5                Depends: libtinyxml2-6a (>= 6.0.0) but it is not installable
#27 469.5 E: Unable to correct problems, you have held broken packages.
``` 

Also, thanks for making the changes, I completely forgot about the PR until I saw the notification just now.

**Comment by lreiher** [comment]
Thanks for getting back to this! We have already integrated the changes because we were in a bit of a hurry to push a new release.

We are now going to merge this and create a new release. Please note that the GitHub action checks have not succeeded, because the required GitHub action secrets are not shared with pull requests from repository forks.

**Pull request merged**